File: tests/rspec/pages/publish/_file_modal.rb
class FileModal < SitePrism::Section
  element :title, 'h1'
  element :upload_button, '.tbl-search a.btn'
  elements :filters, '.filters > ul > li > a.has-sub'
  elements :view_filters, '.filters > ul > li:last-child ul a'
  elements :files, 'table tbody td'
end

File: tests/cypress/cypress/elements/pages/publish/FileModal.js
import ControlPanel from '../ControlPanel'

class FileModal extends ControlPanel {
  constructor() {
    super()
    this.elements({
      'file_modal': '.modal-file',
      'title': '.modal-file .title-bar__title',
      'upload_button': '.modal-file .button.button--primary:contains("Upload")',
      'filters': '.modal-file .filter-search-bar > div > .filter-search-bar__item',
      'view_filters': '.modal-file .filter-search-bar .filter-search-bar__item .filter__viewtype',
      'files': '.modal-file table tbody tr'
    })
  }
}
export default FileModal;
